### ISSUE TITLE
Code for SHOT or TENT for segmentation 

### ISSUE THREAD

**Issue by ChunmengLiu1** [zhang-haojie/wesam#14]
Hi! Thank you for your good work!

Could you provide some code information about SHOT or TENT about weak label segmentation (compared in yours wesam)?
I am looking for semantic segmentation code in TENT, but they didn't release. Thanks for your help!

**Comment by zhang-haojie** [comment]
Since TENT or SHOT are adaptation methods used for classification models, as we are not comfortable publishing these implementations. Here I give the steps of the core implementation of TENT for reference only
```
for pred_mask in pred_masks:
    pred_mask = F.sigmoid(pred_mask)
    pred_mask = torch.clamp(pred_mask, min=0, max=1)

    for mask in pred_mask:
        mask = mask.view(-1)    # [256 * 256]
        loss = - mask.softmax(0) * torch.log(mask.softmax(0)) - (1 - mask.softmax(0)) * torch.log(1 - mask.softmax(0))
        loss_total += loss[~torch.isnan(loss)].sum()
```

**Comment by ChunmengLiu1** [comment]
> Since TENT or SHOT are adaptation methods used for classification models, as we are not comfortable publishing these implementations. Here I give the steps of the core implementation of TENT for reference only
> 
> ```
> for pred_mask in pred_masks:
>     pred_mask = F.sigmoid(pred_mask)
>     pred_mask = torch.clamp(pred_mask, min=0, max=1)
> 
>     for mask in pred_mask:
>         mask = mask.view(-1)    # [256 * 256]
>         loss = - mask.softmax(0) * torch.log(mask.softmax(0)) - (1 - mask.softmax(0)) * torch.log(1 - mask.softmax(0))
>         loss_total += loss[~torch.isnan(loss)].sum()
> ```

Hi, many thanks!

**Comment by Emilychenlin** [comment]
What is the backbone of TENT you used in your paper?

**Comment by zhang-haojie** [comment]
> What is the backbone of TENT you used in your paper?

All SAM encoder

**Comment by Emilychenlin** [comment]
> > What is the backbone of TENT you used in your paper?
> 
> All SAM encoder

Thanks a lot!
